### PR TITLE
ui_tests: extend test_selinuxusermap.py suite

### PR DESCRIPTION
**Pull request by Rezney**
Extend test_selinuxusermap.py suite with new test cases. Details in the ticket.
    
We also modify "add_table_associations" to handle "cancel" and "negative" in the way other methods works.
    
Lastly, we start using dialog_btn=None to test keyboard confirmation as we did use it incorrectly with "Negative=True" where it was already confirmed by "click".

### CODE REVIEW

File: ipatests/test_webui/test_selinuxusermap.py [hunk 1]
```diff
@@ -89,9 +99,22 @@ def test_mod(self):
...
         self.mod_rule_tables(tables, categories, [])
 
+        # add associations then cancel
+        def get_t_vals(t):
+            table = t[0]
+            k = t[1]
+            e = []
+            if len(t) > 2:
+                e = t[2]
+            return table, k, e
```
Comment (pvoborni) [inline ipatests/test_webui/test_selinuxusermap.py]: This helper method seems redundant to me. Can be solved just by:

```
for t in tables:
     table = t[0]
     keys = t[1]        
     self.add_table_associations(table, [keys[0]], confirm_btn='cancel')
```

I also wonder why the same method in `mod_rule_tables` has the `e` in it. But that's another story.

File: ipatests/test_webui/test_selinuxusermap.py [hunk 2]
```diff
@@ -25,22 +25,30 @@
 from ipatests.test_webui.ui_driver import screenshot
 import ipatests.test_webui.data_user as user
 import ipatests.test_webui.data_group as group
+import ipatests.test_webui.data_selinuxusermap as selinux
```
Comment (pvoborni) [inline ipatests/test_webui/test_selinuxusermap.py]: `selinux` is bit misleading. `selinuxmap` or `selinuxusermap` seems better

File: ipatests/test_webui/test_selinuxusermap.py [hunk 3]
```diff
@@ -104,9 +127,245 @@ def test_actions(self):
...
+        # test selinux usermap undo button
+        self.add_record(selinux.ENTITY, selinux.DATA)
+        self.navigate_to_record(selinux.PKEY)
+        self.fill_fields(selinux.DATA['mod'])
+        self.assert_undo_button('description')
+        desc_undo_btn_e = '.widget [name="description"] .btn.undo'
+        desc_undo_btn = self.find(desc_undo_btn_e, By.CSS_SELECTOR)
+        desc_undo_btn.click()
+        self.assert_undo_button('description', visible=False)
+        self.verify_btn_action(mod_description)
```
Comment (pvoborni) [inline ipatests/test_webui/test_selinuxusermap.py]: these 4 'undo' lines look as a perfect candidate for a new ui_driver method. E.g. click_undo(field)

File: ipatests/test_webui/test_selinuxusermap.py (hunk 2 above)
Comment (Rezney) [reply ipatests/test_webui/test_selinuxusermap.py]: Yep, makes sense. Fixed.

File: ipatests/test_webui/test_selinuxusermap.py (hunk 1 above)
Comment (Rezney) [reply ipatests/test_webui/test_selinuxusermap.py]: Fixed.

File: ipatests/test_webui/test_selinuxusermap.py (hunk 3 above)
Comment (Rezney) [reply ipatests/test_webui/test_selinuxusermap.py]: New method introduced.

**Comment by varunmylaraiah** [comment]
@Rezney SelinuxUsermap test failed in "test disable enable multiple SELinux rules"

**Comment by Rezney** [comment]
Aah, I see the problem now. The record is already there so it is firstly deleted and it is throwing one more notification which conflicts with expected one.

**Comment by Rezney** [comment]
master:

* 8d508b8e954aca460949b5151422bf5f59f9eee8 ui_tests: extend test_selinuxusermap.py suite
* 0959c47676d11fa73ab10fb0559f8dc045d7f8dd ui_tests: add click_undo_button() func

**Pull request closed without merging**